en/services/parameter.md

```diff
@@ -98,9 +98,14 @@ The sequence of operations is:
 1. Drone sends all parameters individually in [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
    - The drone should allow a break between each message in order to avoid saturating the link.
 1. GCS accumulates parameters in order to know which parameters have been/not been received  (`PARAM_VALUE` contains total number of params and index of current param).
-1. GCS starts timeout after each `PARAM_VALUE` message in order to detect when parameters are no longer being sent.
-1. After timeout (messages no longer being sent) the GCS can request any missing parameter values by [requesting them individually](#read_single) (using [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ)).
+1. GCS starts timeout after each `PARAM_VALUE` message in order to detect when parameters are no longer being sent.  
+1. After timeout (messages no longer being sent) the GCS can request any missing parameter values.
+ 
+   The GCS determines what parameters are missing based on the `param_count` and `param_index` fields from received [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
+   The messages are [requested individually](#read_single) using [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) and the missing `param_index`
 
+   > **Warning** The sending system must keep the `param_count` and `param_index`s the same during the whole "read all parameters" operation. Outside of this operation the parameter indecies cannot be assumed to be constant, and parameters must be read by id (name).
```

Comment:
Although this is what is needed from a ground station side to implement a robust parameter download there is currently no way for the vehicle side to know when the initial parameter request operation is fully complete.

Reply:
> * Autopilot must maintain the parameter index and count "forever"?

That's not really feasable since ArduPilot already doesn't do that. It's more like the index can't just change randomly without some sort of user input. For example in ArduPilot there are enable params which turn on other sets of params. If you set the enable param, then the indices change to take into account the newly added parameters.

Reply:
It means the protocol can't handle the concept of new parameters showing up randomly. And if they do, things may or may not go wrong from the ground station side. Without a new protocol there isn't much way to make it any clearer.